crates/flux-infer/src/fixpoint_encoding.rs

```diff
@@ -1459,6 +1484,30 @@ impl<'genv, 'tcx> ExprEncodingCtxt<'genv, 'tcx> {
         }
     }
 
+    fn define_const_for_to_int(
+        &mut self,
+        sort_args: &[rty::SortArg],
+        scx: &mut SortEncodingCtxt,
+    ) -> fixpoint::Var {
+        let key = ConstKey::ToInt(sort_args.into());
+        self.const_map
+            .entry(key)
+            .or_insert_with(|| {
+                let fsort = self
+                    .genv
+                    .sort_of_spec_func(&flux_middle::fhir::SpecFuncKind::Cast)
+                    .instantiate(sort_args);
+                let fsort = rty::PolyFuncSort::new(List::empty(), fsort);
+                let sort = scx.func_sort_to_fixpoint(&fsort);
```

Comment:
Since I suggested getting rid of `sort_of_spec_func`, we can inline the function sort here. We are already relying on `cast` having exactly one argument for the encoding, so this is already a leaky abstraction, and we don't lose much with the extra "coordination".